src/functions.php

```diff
@@ -423,13 +423,15 @@ function upload($source, $destination, array $config = [])
     $host = Context::get()->getHost();
     $source = parse($source);
     $destination = parse($destination);
-    $config['options'] = is_array($config['options']) ? $config['options'] : [];
 
     if ($host instanceof Localhost) {
         $rsync->call($host->getHostname(), $source, $destination, $config);
     } else {
         $sshArguments = $host->getSshArguments()->getCliArguments();
         if (empty($sshArguments) === false) {
+            if (!isset($config['options']) || !is_array($config['options'])) {
```

Comment:
It's the same as the original is_array check (line 426), but with an isset before to prevent the notice.

Consider options would contain a string (or another non-array value):
$config['options'] = 'some string';

$config['options][] = "-e 'ssh $sshArguments'";
would cause:
PHP Fatal error:  Uncaught Error: [] operator not supported for strings